tests/integration/PHPUnit/IntegrationMockedTestCase.php

```diff
@@ -0,0 +1,353 @@
+		$token->set_expiry_year('2030'); // Missing expiry year in your original code
+
+		$result = $token->save();
+
+		if (!$result || is_wp_error($result)) {
+			throw new \Exception('Failed to save payment token: ' .
+				(is_wp_error($result) ? $result->get_error_message() : 'Unknown error'));
+		}
+
+		$saved_token = \WC_Payment_Tokens::get($token->get_id());
+		if (!$saved_token || $saved_token->get_id() !== $token->get_id()) {
+			throw new \Exception('Token was not saved correctly');
+		}
+
+		return $token;
+	}
+
+	/**
+	 * Helper method to create a subscription for testing.
+	 *
+	 * @param int $customer_id The customer ID
+	 * @param string $payment_method The payment method
+	 * @param string $sku
+	 * @return WC_Subscription
+	 * @throws \WC_Data_Exception
+	 */
+	public function createSubscription(int $customer_id = 1, string $payment_method = 'ppcp-gateway', $sku = 'DUMMY SUB SKU'): WC_Subscription
+	{
+		// Create a product if not provided
+		$product_id = $this->createAProductIfNotProvided($sku);
+
+		$order = $this->getMockedOrder($customer_id, $payment_method, $product_id, $set_paid = true);
+
+		$subscription = new WC_Subscription();
+		$subscription->set_customer_id($customer_id);
+		$subscription->set_payment_method($payment_method);
+		$subscription->set_status('active');
+		$subscription->set_parent_id($order->get_id());
+		$subscription->set_billing_period('month');
+		$subscription->set_billing_interval(1);
+
+		// Add a product to the subscription
+		$subscription_item = new WC_Order_Item_Product();
+		$subscription_item->set_props([
+			'product_id' => $product_id,
+			'quantity' => 1,
+			'subtotal' => 10,
+			'total' => 10,
+		]);
+		$subscription->add_item($subscription_item);
+		$subscription->set_date_created(current_time('mysql'));
+		$subscription->set_start_date(current_time('mysql'));
+		$subscription->set_next_payment_date(date('Y-m-d H:i:s', strtotime('+1 month', current_time('timestamp'))));
+		$subscription->save();
+
+		return $subscription;
+	}
+
+	/**
+	 * Creates a renewal order for testing
+	 *
+	 * @param int $customer_id
+	 * @param string $gateway_id
+	 * @param int $subscription_id
+	 * @return WC_Order
+	 */
+	protected function createRenewalOrder(int $customer_id, string $gateway_id, int $subscription_id): WC_Order
+	{
+		$renewal_order = $this->getMockedOrder($customer_id, $gateway_id, $this->default_product_id, false);
+		$renewal_order->update_meta_data('_subscription_renewal', $subscription_id);
+		$renewal_order->save();
+
+		return $renewal_order;
+	}
+
+	/**
+	 * Mocks the OrderEndpoint to return a successful/failed order.
+	 *
+	 * @param string $intent The order intent (CAPTURE or AUTHORIZE)
+	 * @param bool $success Whether the order was successful
+	 * @return object The mocked OrderEndpoint
+	 */
+	public function mockOrderEndpoint(string $intent = 'CAPTURE', bool $success = true): object
+	{
+		$order_endpoint = $this->getMockBuilder(OrderEndpoint::class)
```

Comment:
Looks like we were using Mockery in most of the tests in this project. Should we maybe continue that for consistency, or should we switch to PHPUnit mocks? I am not sure if they have any significant difference.